### ISSUE TITLE
Buttons to donate are inaccessible with screen reader

### ISSUE THREAD

**Issue by aardrian** [opencollective/opencollective#3790]
**Describe the bug**
The pledge / contribution detail buttons do not convey their group name (that one set is a pledge amount, that another is a recurrence), state (pressed or not), properties (required), nor their role beyond buttons (that they are essentially a group of toggle buttons). Nor does the select-like control to choose a % to donate on top of the pledge.

**To Reproduce**
1. Open your favorite screen reader (JAWS, NVDA, VoiceOver, TalkBack, Orca, Narrator);
2. Go to https://opencollective.com/focus-visible-in-webkit/contribute/donation-23724/checkout;
3. Navigate to the controls immediately following the heading _Contribution details_;
4. Observe there is no group name nor announcement that one is pressed or not;
5. Hit `Enter` on _One Time_, _Monthly_, or _Yearly_;
6. Observe the message dialog is not announced;
7. Observe the dialog allows focus behind it and does not go away;
8. Navigate to the dollar value buttons;
9. Observe there is no group name nor announcement that a button is pressed or not;
10. Press `Tab` and find you are in an unlabeled control (in JAWS you might here only "read only edit,
Use your reading keys to read the text.").

**Expected behavior**
For each button I expect to hear its group name ("Recurrence" or "Donation Amount"), whether or not it is active (via aria-pressed), whether the group is required, and if something is disallowed.

For the modal dialog I expect focus to be moved to it, the background to be restricted from receiving focus, and it to be announced. When closed I expect focus to be returned where it was.

For the select-like control, I expect to hear its label (accessible name), its value, the control type, and if it is required.

**Screenshots**
See the linked issue.

**Desktop (please complete the following information):**
 - OS: Windows, Mac
 - Browser: Chrome, Safari, Firefox, Edge.
 - Version: Latest release of each

**Smartphone (please complete the following information):**
 - Device: iPhone 8, Pixel 2
 - OS: iOS 14.1, Android 11
 - Browser: Safari, Firefox, Chrome
 - Version: Latest release of each

**Additional context**
Combined, these issues (and some other obvious ones) on this one screen amount to WCAG failures that correspond to:
* 1.3.1 Info and Relationships
* 1.4.1 Use of Color (the pressed state is conveyed only by blueness)
* 1.4.3 Contrast (Minimum)
* 1.4.4 Resize text (see the linked issue below)
* 1.4.11 Non-text Contrast (bounding areas)
* 2.4.6 Headings and Labels
* 3.3.1 Error Identification
* 3.3.2 Labels or Instructions
* 4.1.2 Name, Role, Value

This essentially re-creates and expands my notes from [#5196 Pledge amounts a11y issue/clarity](https://github.com/opencollective/opencollective-frontend/issues/5196), which was closed for not following the rules to file in this repo instead of the one related to the screens with issues.

The simplest fix might be to use radio buttons. You can still style them to look like what you have, but they bring all the accessibility information along for free.

If you keep the buttons, the group of buttons needs a group label, the buttons need to indicate their toggleness, and the number input should be selectable with mouse, not just keyboard.

An example structure (for the pledge page, referenced in the linked issue):
```
<div role="group" aria-labelledby="GrpLbl">
 <label for="amount" id="GrpLbl">Amount (USD)</label>
 <button aria-pressed="false">$5</button>
 <button aria-pressed="false">$10</button>
 <button aria-pressed="false">$15</button>
 <button aria-pressed="false">$20</button>
 <button aria-pressed="false">$50</button>
 <button aria-pressed="true">$100</button>
 <button aria-pressed="false">$250</button>
 <input type="number" id="amount" value="100">
</div>
```
You can hang styles off `button[aria-pressed="true"]` to ensure the visual style matches the programmatic state.

The `group` role will announce to a screen reader that the buttons are related and provide a group label for them (via `aria-labelledby`).

Given how many people see the `:focus-visible` feature as an accessibility feature, I have a genuine concern some people are unable to donate because this process is inaccessible.

**Comment by aardrian** [comment]
Please add the labels `frontend` and `accessibility`.

**Comment by aardrian** [comment]
I appreciate the other labels have been added. However, this is a bug report, not a request for an enhancement.

This one part of the screen fails WCAG in at least 9 different ways, which is illegal in some jurisdictions and also a lawsuit risk. Because some of the issues are blocking issues, preventing a user from completing the task, they would be prioritized as _critical_ in an accessibility evaluation.

Please restore the `bug` label.

**Comment by Betree** [comment]
> I appreciate the other labels have been added. However, this is a bug report, not a request for an enhancement.
> 
> This one part of the screen fails WCAG in at least 9 different ways, which is illegal in some jurisdictions and also a lawsuit risk. Because some of the issues are blocking issues, preventing a user from completing the task, they would be prioritized as _critical_ in an accessibility evaluation.
> 
> Please restore the `bug` label.

@aardrian this classification is just a way for us to help triage issues, putting that along with `enhancement` was part of routine triaging. It doesn't reflect any kind or prioritization, nor does it implies that we don't want to have this fixed.

After a quick internal discussion, we've agreed to move to `bug` so there's no confusion on this in the future.

**Comment by Betree** [comment]
And thanks for opening this very detailed issue @aardrian, it really helps!

**Comment by SudharakaP** [comment]
Will be working on this one. 😄

**Comment by Betree** [comment]
:trophy: This issue has been completed by @SudharakaP. To claim the **$300** bounty, you can submit an invoice to https://opencollective.com/engineering/expenses/new (please mention the issue number).

This first batch includes screen readers compatibility for the mentioned components, as well as global fixes for the accessibility of the modals (especially regarding focus) and various enhancements. It should be deployed in the next few days.

The PR did **not** include fixes for the contrast issues. @opencollective/design has planned a project for this sprint to review and fix the contrast issues globally, not only in the contribution flow.

**Comment by aardrian** [comment]
Ace!

When will these changes deploy?

**Comment by Betree** [comment]
> Ace!
> 
> When will these changes deploy?

As mentioned, it should be deployed in the next few days.
